packages/client/test/sim/beaconsync.spec.ts

```diff
@@ -0,0 +1,221 @@
+      EOATransferToBalance += 1000000n
+
+      balance = await client.request('eth_getBalance', [EOATransferToAccount, 'latest'])
+      assert.equal(BigInt(balance.result), EOATransferToBalance, 'sent a simple ETH transfer 2x')
+
+      balance = await client.request('eth_getBalance', [sender, 'latest'])
+      assert.ok(
+        balance.result !== undefined,
+        'remaining sender balance after transfers and gas fee'
+      )
+    },
+    2 * 60_000
+  )
+
+  it.skipIf(process.env.BEACON_SYNC === undefined)(
+    'setup beacon sync',
+    async () => {
+      // start client inline here for beacon sync
+      const peerBeaconUrl = 'http://127.0.0.1:9596'
+      const {
+        ejsInlineClient,
+        peerConnectedPromise,
+        beaconSyncRelayer: relayer,
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+      } = (await createBeaconSyncClient(
+        common,
+        customGenesisState,
+        [nodeInfo.enode],
+        peerBeaconUrl
+      ).catch((e) => {
+        console.log(e)
+        return null
+      })) ?? {
+        ejsInlineClient: null,
+        peerConnectedPromise: Promise.reject('Client creation error'),
+      }
+      ejsClient = ejsInlineClient
+      beaconSyncRelayer = relayer
+      assert.ok(ejsClient !== null, 'ethereumjs client started')
+
+      const enode = (ejsClient!.server('rlpx') as RlpxServer)!.getRlpxInfo().enode
+      const res = await client.request('admin_addPeer', [enode])
+      assert.equal(res.result, true, 'successfully requested Geth add EthereumJS as peer')
+
+      const peerConnectTimeout = new Promise((_resolve, reject) => setTimeout(reject, 10000))
+      try {
+        await Promise.race([peerConnectedPromise, peerConnectTimeout])
+        assert.ok(true, 'connected to geth peer')
+      } catch (e) {
+        console.log(e)
+        assert.fail('could not connect to geth peer in 10 seconds')
+      }
+    },
+    60_000
+  )
+
+  it.skipIf(process.env.BEACON_SYNC === undefined)(
+    'should beacon sync and finish',
+    async () => {
+      if (ejsClient !== null && beaconSyncRelayer !== null) {
+        // wait on the sync promise to complete if it has been called independently
+        const syncTimeout = new Promise((_resolve, reject) => setTimeout(reject, 8 * 60_000))
+        const beaconSyncPromise = beaconSyncRelayer.start()
+
+        try {
+          // call sync if not has been called yet
+          void ejsClient.services[0].synchronizer?.sync()
+          const syncResponse = await Promise.race([beaconSyncPromise, syncTimeout])
+          assert(syncResponse.syncState, 'SYNCED', 'beaconSyncRelayer should have synced client')
+          await ejsClient.stop()
+          assert.ok(true, 'completed beacon sync')
+        } catch (e) {
+          assert.fail('could not complete beacon sync in 8 minutes')
+        }
+      } else {
+        assert.fail('ethereumjs client not setup properly for snap sync')
```

Comment:
What is this error referencing?  Is snap sync somehow being tested here?

Reply:
sorry copied over from snapsyncspec , will cleanup